torch/distributed/optim/apply_optimizer_in_backward.py

```diff
@@ -88,3 +88,26 @@ def optimizer_hook(*_unused) -> None:
 
     for param in params:
         _apply_optimizer_in_backward_to_param(param)
+
+
+def _get_optimizers_in_backward(model: torch.nn.Module) -> List[torch.optim.Optimizer]:
+    """
+    Return a list of in-backward optimizers applied to ``model``'s parameters. Note that these
+    optimizers are not intended to directly have their ``step`` or ``zero_grad`` methods called
+    by the user and are intended to be used for things like checkpointing.
+
+    Args:
+        model: (torch.nn.Module): model to retrieve in-backward optimizers for
+
+    Returns:
+        List[torch.optim.Optimizer]: the in-backward optimizers.
+
+    Example::
+        _apply_optimizer_in_backward(torch.optim.SGD, model.parameters(), {'lr': 0.01})
+        optims = _get_optimizers_in_backward(model)
+    """
+    optims = []
+    for param in model.parameters():
+        optims.extend(param._in_backward_optimizers)
```

Comment:
Should we use `getattr` and also check if `_in_backward_optimizers` exists in the param?